drivers/SmartThings/sonos/src/sonos_driver.lua

```diff
@@ -384,6 +337,35 @@ function SonosDriver:get_oauth_token()
   return nil, "no token"
 end
 
+function SonosDriver:wait_for_oauth_token(timeout)
```

Comment:
Can you add some comments on the logic here for the non initiated with sonos and how the oauth works? The `maybe_token` looks to be the case when auth is already setup and we have a non expired token, but the `subscribe`, `receive` and then re-calling of `get_oauth_token` isnt obvious what that is accomplishing.